zerver/lib/actions.py

```diff
@@ -4518,6 +4533,24 @@ def do_update_message(user_profile: UserProfile, message: Message,
         )
         changed_messages += messages_list
 
+        if new_stream is not None:
+            assert stream_being_edited is not None
+            UserMessage.objects.filter(
+                user_profile__in=[sub.user_profile for sub in guest_subs],
+                message__in=changed_messages,
+            ).delete()
+
+            delete_event: Dict[str, Any] = {
+                'type': 'delete_message',
+                'message_ids': [msg.id for msg in changed_messages],
+                'message_type': 'stream',
+                'stream_id': stream_being_edited.id,
+                'topic': orig_topic_name,
+            }
+            delete_event_notify_user_ids = [user.user_profile_id for user in guest_subs]
```

Comment:
`s/user/sub/`; we don't use `user` as a local variable except for UserProfile objects.